cerberus_anymal_control_common/cerberus_anymal_control_base/include/cerberus_anymal_control_base/CerberusAnymalControlBaseRos.tpp

```diff
@@ -0,0 +1,118 @@
+    // Publish the control commands
+    RosInterface::advanceRos();
+  }
+}
+
+template<typename ConcreteQuadrupedController>
+void CerberusAnymalControlBaseRos<ConcreteQuadrupedController>::commandVelCallback(const geometry_msgs::Twist &twistMsg) {
+  boost::shared_lock<boost::shared_mutex>lock(Controller::data_mutex_);
+  // Linear Velocity
+  Controller::desLinearVelocity_[0] = twistMsg.linear.x;
+  Controller::desLinearVelocity_[1] = twistMsg.linear.y;
+  Controller::desLinearVelocity_[2] = twistMsg.linear.z;
+  // Angualr Velocity
+  Controller::desAngularVelocity_[0] = twistMsg.angular.x;
+  Controller::desAngularVelocity_[1] = twistMsg.angular.y;
+  Controller::desAngularVelocity_[2] = twistMsg.angular.z;
+}
+
+template<typename ConcreteQuadrupedController>
+void CerberusAnymalControlBaseRos<ConcreteQuadrupedController>::jointStateCallback(const sensor_msgs::JointState &jointStateMsg) {
+  boost::shared_lock<boost::shared_mutex>lock(Controller::data_mutex_);
+  // LF
+  Controller::genCoordinates_(7) = jointStateMsg.position[0];
+  Controller::genVelocities_(6) = jointStateMsg.velocity[0];
+  Controller::genCoordinates_(8) = jointStateMsg.position[1];
+  Controller::genVelocities_(7) = jointStateMsg.velocity[1];
+  Controller::genCoordinates_(9) = jointStateMsg.position[2];
+  Controller::genVelocities_(8) = jointStateMsg.velocity[2];
+  // LH
+  Controller::genCoordinates_(13) = jointStateMsg.position[3];
+  Controller::genVelocities_(12) = jointStateMsg.velocity[3];
+  Controller::genCoordinates_(14) = jointStateMsg.position[4];
+  Controller::genVelocities_(13) = jointStateMsg.velocity[4];
+  Controller::genCoordinates_(15) = jointStateMsg.position[5];
+  Controller::genVelocities_(14) = jointStateMsg.velocity[5];
+  // RF
+  Controller::genCoordinates_(10) = jointStateMsg.position[6];
+  Controller::genVelocities_(9) = jointStateMsg.velocity[6];
+  Controller::genCoordinates_(11) = jointStateMsg.position[7];
+  Controller::genVelocities_(10) = jointStateMsg.velocity[7];
+  Controller::genCoordinates_(12) = jointStateMsg.position[8];
+  Controller::genVelocities_(11) = jointStateMsg.velocity[8];
+  // RH
+  Controller::genCoordinates_(16) = jointStateMsg.position[9];
+  Controller::genVelocities_(15) = jointStateMsg.velocity[9];
+  Controller::genCoordinates_(17) = jointStateMsg.position[10];
+  Controller::genVelocities_(16) = jointStateMsg.velocity[10];
+  Controller::genCoordinates_(18) = jointStateMsg.position[11];
+  Controller::genVelocities_(17) = jointStateMsg.velocity[11];
+  // Set the flag
+  if(!Controller::hasJointPositionData_) {
+    Controller::hasJointPositionData_ = true;
+  }
+}
+
+template<typename ConcreteQuadrupedController>
+void CerberusAnymalControlBaseRos<ConcreteQuadrupedController>::bodyTwistCallback(const geometry_msgs::Twist &bodyTwistMsg) {
+  boost::shared_lock<boost::shared_mutex>lock(Controller::data_mutex_);
+  // Set the body twist
+  Controller::genVelocities_(0) = bodyTwistMsg.linear.x;
```

Comment:
you set `genVelocities_(i)` in multiple places by adding single assignments. Consider to have an helper function that takes an input twist vector and sets genVelocities_. This function can be used in multiple places in this PR